services/HttpService/HttpService.ts

```diff
@@ -11,11 +35,95 @@ export class HttpService {
       baseURL: appConfig.apiUrl,
       ...config,
     });
+
+    this.client.interceptors.response.use(response => {
+      const responseConfig: CustomAxiosRequestConfig = response.config;
+
+      if (!responseConfig.responseMapper) {
+        return response;
+      }
+
+      switch (responseConfig.responseMapper.name) {
+        case API_MAPPERS.mapDaoDTOtoDao:
+          response.data = mapDaoDTOtoDao(response.data);
+          break;
+        case API_MAPPERS.mapDaoFeedItemResponseToDaoFeedItemList:
+          if (Array.isArray(response.data.data)) {
```

Comment:
Can we do this check (and all similar checks from this file) and apply proper processing in mapper?

Reply:
I think the mapper function shouldn't decide by incoming data what it should return. Maybe a better solution is if we create a separate type for each condition. What do you think?